### PR TITLE
Update HF Croissant to Croissant 1.0

### PR DESCRIPTION
**Pull request by ccl-core**

### CODE REVIEW

File: services/api/src/api/routes/croissant.py [hunk 1]
```diff
@@ -122,39 +128,55 @@ def get_croissant_from_dataset_infos(
...
-                    }
+                    _remove_none_values(
+                        {
+                            "@type": "cr:Field" if is_v1 else "ml:Field",
+                            "@id": field_name if is_v1 else None,
+                            "name": field_name,
+                            "description": f"ClassLabel column '{column}' from the Hugging Face parquet file.\nLabels:\n"
+                            + ", ".join(f"{name} ({i})" for i, name in enumerate(feature.names)),
+                            "dataType": "sc:Integer",
+                            "source": {"fileSet": {"@id": distribution_name}, "extract": {"column": column}},
```
Comment (marcenacp) [inline services/api/src/api/routes/croissant.py:177]: Distinguish `is_v1`?

File: services/api/src/api/routes/croissant.py [hunk 2]
```diff
@@ -232,6 +275,14 @@ def _get_full_jsonld_parameter(request: Request) -> bool:
     return True
 
 
+def _get_isV1(request: Request) -> bool:
```
Comment (marcenacp) [inline services/api/src/api/routes/croissant.py:278]: `_get_is_v1`?

File: services/api/src/api/routes/croissant.py (hunk 1 above)
Comment (ccl-core) [reply services/api/src/api/routes/croissant.py]: Done.

File: services/api/src/api/routes/croissant.py (hunk 2 above)
Comment (ccl-core) [reply services/api/src/api/routes/croissant.py]: Done.

**Review by marcenacp — APPROVED** [review]

**Comment by ccl-core** [comment]
Hello @severo , this PR is for updating the Croissant dataset definitions for HF to conform to the latest [Croissant 1.0 specs](https://mlcommons.github.io/croissant/docs/croissant-spec.html). It will still be possible to generate 0.8-compliant Croissant definitions using the `is_V1` flag.

Thanks!

**Comment by severo** [comment]
for reference: previous PR https://github.com/huggingface/datasets-server/pull/2535

**Comment by severo** [comment]
I think we don't need to keep compatibility with previous croissant versions, we can only support v1

**Comment by ccl-core** [comment]
This sounds good to me. @marcenacp do you agree? Just to make sure we are on the same page before proceeding

**Comment by marcenacp** [comment]
@ccl-core @severo We wanted to make sure v1 is OK (e.g., using the [health tool](https://github.com/mlcommons/croissant/tree/main/health#croissant--online-health)) before toggling. If you think there is no risk, LGTM to directly support v1.

**Comment by severo** [comment]
yes, I think we will iterative quicker if we don't manage multiple versions

**Comment by ccl-core** [comment]
Sounds good! The code is also much more readable this way.

**Review by severo — COMMENTED** [review]
Sounds good for me! Please fix the quality issue (two f-strings -> strings), and we can merge

**Comment by ccl-core** [comment]
Thanks! Hopefully it should be fixed now :)

**Comment by severo** [comment]
Good contribution @ccl-core! I'll ping you when it's deployed and you can try in production.

**Comment by severo** [comment]
https://datasets-server.huggingface.co/croissant?dataset=mnist

before:

```json
{"@context":{"@language":"en","@vocab":"https://schema.org/","column":"ml:column","data":{"@id":"ml:data","@type":"@json"},"dataType":{"@id":"ml:dataType","@type":"@vocab"},"extract":"ml:extract","field":"ml:field","fileProperty":"ml:fileProperty","format":"ml:format","includes":"ml:includes","isEnumeration":"ml:isEnumeration","jsonPath":"ml:jsonPath","ml":"http://mlcommons.org/schema/","parentField":"ml:parentField","path":"ml:path","recordSet":"ml:recordSet","references":"ml:references","regex":"ml:regex","repeated":"ml:repeated","replace":"ml:replace","sc":"https://schema.org/","separator":"ml:separator","source":"ml:source","subField":"ml:subField","transform":"ml:transform"},"@type":"sc:Dataset","name":"mnist","description":"mnist dataset hosted on Hugging Face and contributed by the HF Datasets community","license":"","url":"https://huggingface.co/datasets/mnist","distribution":[{"@type":"sc:FileObject","name":"repo","description":"The Hugging Face git repository.","contentUrl":"https://huggingface.co/datasets/mnist/tree/refs%2Fconvert%2Fparquet","encodingFormat":"git+https","sha256":"https://github.com/mlcommons/croissant/issues/80"},{"@type":"sc:FileSet","name":"parquet-files-for-config-mnist","containedIn":"repo","encodingFormat":"application/x-parquet","includes":"mnist/*/*.parquet"}],"recordSet":[{"@type":"ml:RecordSet","name":"record_set_mnist","description":"mnist - 'mnist' subset\n\nAdditional information:\n- 2 splits: train, test","field":[{"@type":"ml:Field","name":"image","description":"Image column 'image' from the Hugging Face parquet file.","dataType":"sc:ImageObject","source":{"distribution":"parquet-files-for-config-mnist","extract":{"column":"image"},"transform":{"jsonPath":"bytes"}}},{"@type":"ml:Field","name":"label","description":"ClassLabel column 'label' from the Hugging Face parquet file.\nLabels:\n0 (0), 1 (1), 2 (2), 3 (3), 4 (4), 5 (5), 6 (6), 7 (7), 8 (8), 9 (9)","dataType":"sc:Integer","source":{"distribution":"parquet-files-for-config-mnist","extract":{"column":"label"}}}]}]}
```

after:

```json
{"@context":{"@language":"en","@vocab":"https://schema.org/","citeAs":"cr:citeAs","column":"cr:column","conformsTo":"dct:conformsTo","cr":"http://mlcommons.org/croissant/","data":{"@id":"cr:data","@type":"@json"},"dataBiases":"cr:dataBiases","dataCollection":"cr:dataCollection","dataType":{"@id":"cr:dataType","@type":"@vocab"},"dct":"http://purl.org/dc/terms/","extract":"cr:extract","field":"cr:field","fileProperty":"cr:fileProperty","fileObject":"cr:fileObject","fileSet":"cr:fileSet","format":"cr:format","includes":"cr:includes","isEnumeration":"cr:isEnumeration","isLiveDataset":"cr:isLiveDataset","jsonPath":"cr:jsonPath","key":"cr:key","md5":"cr:md5","parentField":"cr:parentField","path":"cr:path","personalSensitiveInformation":"cr:personalSensitiveInformation","recordSet":"cr:recordSet","references":"cr:references","regex":"cr:regex","repeated":"cr:repeated","replace":"cr:replace","sc":"https://schema.org/","separator":"cr:separator","source":"cr:source","subField":"cr:subField","transform":"cr:transform"},"@type":"sc:Dataset","name":"mnist","conformsTo":"http://mlcommons.org/croissant/1.0","description":"mnist dataset hosted on Hugging Face and contributed by the HF Datasets community","license":"","url":"https://huggingface.co/datasets/mnist","distribution":[{"@type":"sc:FileObject","@id":"repo","name":"repo","description":"The Hugging Face git repository.","contentUrl":"https://huggingface.co/datasets/mnist/tree/refs%2Fconvert%2Fparquet","encodingFormat":"git+https","sha256":"https://github.com/mlcommons/croissant/issues/80"},{"@type":"sc:FileSet","@id":"parquet-files-for-config-mnist","name":"parquet-files-for-config-mnist","containedIn":{"@id":"repo"},"encodingFormat":"application/x-parquet","includes":"mnist/*/*.parquet"}],"recordSet":[{"@type":"cr:RecordSet","@id":"record_set_mnist","name":"record_set_mnist","description":"mnist - 'mnist' subset\n\nAdditional information:\n- 2 splits: train, test","field":[{"@type":"cr:Field","@id":"image","name":"image","description":"Image column 'image' from the Hugging Face parquet file.","dataType":"sc:ImageObject","source":{"fileSet":{"@id":"parquet-files-for-config-mnist"},"extract":{"column":"image"},"transform":{"jsonPath":"bytes"}}},{"@type":"cr:Field","@id":"label","name":"label","description":"ClassLabel column 'label' from the Hugging Face parquet file.\nLabels:\n0 (0), 1 (1), 2 (2), 3 (3), 4 (4), 5 (5), 6 (6), 7 (7), 8 (8), 9 (9)","dataType":"sc:Integer","source":{"fileSet":{"@id":"parquet-files-for-config-mnist"},"extract":{"column":"label"}}}]}]}
```

Metadata in the header of https://huggingface.co/datasets/mnist:

before:

```json
{
  "@context": {
    "@language": "en",
    "@vocab": "https://schema.org/",
    "column": "ml:column",
    "data": {
      "@id": "ml:data",
      "@type": "@json"
    },
    "dataType": {
      "@id": "ml:dataType",
      "@type": "@vocab"
    },
    "extract": "ml:extract",
    "field": "ml:field",
    "fileProperty": "ml:fileProperty",
    "format": "ml:format",
    "includes": "ml:includes",
    "isEnumeration": "ml:isEnumeration",
    "jsonPath": "ml:jsonPath",
    "ml": "http://mlcommons.org/schema/",
    "parentField": "ml:parentField",
    "path": "ml:path",
    "recordSet": "ml:recordSet",
    "references": "ml:references",
    "regex": "ml:regex",
    "repeated": "ml:repeated",
    "replace": "ml:replace",
    "sc": "https://schema.org/",
    "separator": "ml:separator",
    "source": "ml:source",
    "subField": "ml:subField",
    "transform": "ml:transform"
  },
  "@type": "sc:Dataset",
  "distribution": [
    {
      "@type": "sc:FileObject",
      "name": "repo",
      "description": "The Hugging Face git repository.",
      "contentUrl": "https://huggingface.co/datasets/mnist/tree/refs%2Fconvert%2Fparquet",
      "encodingFormat": "git+https",
      "sha256": "https://github.com/mlcommons/croissant/issues/80"
    },
    {
      "@type": "sc:FileSet",
      "name": "parquet-files-for-config-mnist",
      "containedIn": "repo",
      "encodingFormat": "application/x-parquet",
      "includes": "mnist/*/*.parquet"
    }
  ],
  "recordSet": [
    {
      "@type": "ml:RecordSet",
      "name": "record_set_mnist",
      "description": "mnist - 'mnist' subset\n\nAdditional information:\n- 2 splits: train, test",
      "field": [
        {
          "@type": "ml:Field",
          "name": "image",
          "description": "Image column 'image' from the Hugging Face parquet file.",
          "dataType": "sc:ImageObject",
          "source": {
            "distribution": "parquet-files-for-config-mnist",
            "extract": {
              "column": "image"
            },
            "transform": {
              "jsonPath": "bytes"
            }
          }
        },
        {
          "@type": "ml:Field",
          "name": "label",
          "description": "ClassLabel column 'label' from the Hugging Face parquet file.\nLabels:\n0 (0), 1 (1), 2 (2), 3 (3), 4 (4), 5 (5), 6 (6), 7 (7), 8 (8), 9 (9)",
          "dataType": "sc:Integer",
          "source": {
            "distribution": "parquet-files-for-config-mnist",
            "extract": {
              "column": "label"
            }
          }
        }
      ]
    }
  ],
  "name": "mnist",
  "description": "The MNIST dataset consists of 70,000 28x28 black-and-white images in 10 classes (one for each digits), with 7,000\nimages per class. There are 60,000 training images and 10,000 test images.",
  "alternateName": [
    "MNIST"
  ],
  "creator": {
    "@type": "Organization",
    "name": "The HF Datasets community",
    "url": "https://github.com/huggingface/datasets"
  },
  "keywords": [
    "image-classification",
    "multi-class-image-classification",
    "expert-generated",
    "found",
    "monolingual",
    "10K&lt;n&lt;100K",
    "extended|other-nist",
    "English",
    "mit",
    "Croissant",
    "🇺🇸 Region: US"
  ],
  "license": "https://choosealicense.com/licenses/mit/",
  "sameAs": "http://yann.lecun.com/exdb/mnist/",
  "url": "https://huggingface.co/datasets/mnist"
}
```

after:

```json
{
  "@context": {
    "@language": "en",
    "@vocab": "https://schema.org/",
    "citeAs": "cr:citeAs",
    "column": "cr:column",
    "conformsTo": "dct:conformsTo",
    "cr": "http://mlcommons.org/croissant/",
    "data": {
      "@id": "cr:data",
      "@type": "@json"
    },
    "dataBiases": "cr:dataBiases",
    "dataCollection": "cr:dataCollection",
    "dataType": {
      "@id": "cr:dataType",
      "@type": "@vocab"
    },
    "dct": "http://purl.org/dc/terms/",
    "extract": "cr:extract",
    "field": "cr:field",
    "fileProperty": "cr:fileProperty",
    "fileObject": "cr:fileObject",
    "fileSet": "cr:fileSet",
    "format": "cr:format",
    "includes": "cr:includes",
    "isEnumeration": "cr:isEnumeration",
    "isLiveDataset": "cr:isLiveDataset",
    "jsonPath": "cr:jsonPath",
    "key": "cr:key",
    "md5": "cr:md5",
    "parentField": "cr:parentField",
    "path": "cr:path",
    "personalSensitiveInformation": "cr:personalSensitiveInformation",
    "recordSet": "cr:recordSet",
    "references": "cr:references",
    "regex": "cr:regex",
    "repeated": "cr:repeated",
    "replace": "cr:replace",
    "sc": "https://schema.org/",
    "separator": "cr:separator",
    "source": "cr:source",
    "subField": "cr:subField",
    "transform": "cr:transform"
  },
  "@type": "sc:Dataset",
  "distribution": [
    {
      "@type": "sc:FileObject",
      "@id": "repo",
      "name": "repo",
      "description": "The Hugging Face git repository.",
      "contentUrl": "https://huggingface.co/datasets/mnist/tree/refs%2Fconvert%2Fparquet",
      "encodingFormat": "git+https",
      "sha256": "https://github.com/mlcommons/croissant/issues/80"
    },
    {
      "@type": "sc:FileSet",
      "@id": "parquet-files-for-config-mnist",
      "name": "parquet-files-for-config-mnist",
      "containedIn": {
        "@id": "repo"
      },
      "encodingFormat": "application/x-parquet",
      "includes": "mnist/*/*.parquet"
    }
  ],
  "recordSet": [
    {
      "@type": "cr:RecordSet",
      "@id": "record_set_mnist",
      "name": "record_set_mnist",
      "description": "mnist - 'mnist' subset\n\nAdditional information:\n- 2 splits: train, test",
      "field": [
        {
          "@type": "cr:Field",
          "@id": "image",
          "name": "image",
          "description": "Image column 'image' from the Hugging Face parquet file.",
          "dataType": "sc:ImageObject",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-mnist"
            },
            "extract": {
              "column": "image"
            },
            "transform": {
              "jsonPath": "bytes"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "label",
          "name": "label",
          "description": "ClassLabel column 'label' from the Hugging Face parquet file.\nLabels:\n0 (0), 1 (1), 2 (2), 3 (3), 4 (4), 5 (5), 6 (6), 7 (7), 8 (8), 9 (9)",
          "dataType": "sc:Integer",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-mnist"
            },
            "extract": {
              "column": "label"
            }
          }
        }
      ]
    }
  ],
  "name": "mnist",
  "description": "The MNIST dataset consists of 70,000 28x28 black-and-white images in 10 classes (one for each digits), with 7,000\nimages per class. There are 60,000 training images and 10,000 test images.",
  "alternateName": [
    "MNIST"
  ],
  "creator": {
    "@type": "Organization",
    "name": "The HF Datasets community",
    "url": "https://github.com/huggingface/datasets"
  },
  "keywords": [
    "image-classification",
    "multi-class-image-classification",
    "expert-generated",
    "found",
    "monolingual",
    "10K&lt;n&lt;100K",
    "extended|other-nist",
    "English",
    "mit",
    "Croissant",
    "🇺🇸 Region: US"
  ],
  "license": "https://choosealicense.com/licenses/mit/",
  "sameAs": "http://yann.lecun.com/exdb/mnist/",
  "url": "https://huggingface.co/datasets/mnist"
}
```

**Comment by ccl-core** [comment]
Hi @severo ! Thank you, I gave a look at the output and it looks like the field names could be improved to actually comply with the latest specs (https://mlcommons.github.io/croissant/docs/croissant-spec.html). Sorry for not noticing that before!

I opened https://github.com/huggingface/datasets-server/pull/2609

I also discussed a few other changes with Omar which are reflected in the PR as well.
After that, I think the output should be fine -- we will run some more extensive testing over the next few days.

**Pull request merged**